Issue 49: adding old SolCache JS sample

aaron-613 opened the pull request:
@TamimiGitHub could probably use your help here..?  @solace-dmckay dug up the old SolCache JS sample that used to be included with the old (pre-Guaranteed-messaging?) API.  I've pushed it into a new branch, and also included a rough working sample I hacked together from the TopicSubscriber sample.

Looking briefly at the code, the old SolCache sample uses jquery, and I don't think our new ones do.  Perhaps this is something we can get the API team to help with, but thought I'd give you a crack first..?

aaron-613 committed:
adding old SolCache JS sample

aaron-613 committed:
adding in my super rough hack samples that works

aaron-613 commented:
⚠ NOTE: this branch is not ready to be MERGED, at all.  Needs some work.  Just checking in for collaboration.

TamimiGitHub commented:
A couple of notes: 
- Not sure why you included `sol-cache.js` and `sol-cache-ui.js`? couldnt you just use the createCacheSession from the solace client API
```
var cs = client.session.createCacheSession({
          cacheName: 'dc1',
          maxAgeSec: 25000,
          maxMessages: 1,
          timeoutMsec: 10000 
        });
```
- What old SolCache sample are you referring to? We dont need `jquery` for a simple html landing page. The other samples here dont use it 

Happy to look at this further

aaron-613 commented:
Hey.  Doing cleanup and stuff.
- the old SolCache sample that Dave was asking for used to be bundled with the API years ago.  It's the `index.html`, `sol-cache*.js`.  It's super old, uses jQuery, looks super out-of-date.
- the `working-rough-sample.html` I made.  But it's not pretty

All the newer JS samples use basic HTML & JS, and `pure.css` for styling.  Wondering if we go about that route, trying to hack/salvage the GUI elements and put them into the same format as other other newer samples?

aaron-613 committed:
added old jQuery libs for solcache sample

aaron-613 commented:
Hey @TamimiGitHub , back on this, trying to clean it up.  I added a bunch of old helper libs (jQuery and other styling libs) from a very old JS API tarball, hoping that it would "just work'.  But still no go.  Now getting:
![image](https://github.com/user-attachments/assets/56346664-4b79-442d-92f9-70d69e63d6bf)

Urgh.  Don't want to spend a lot of time on this just to get it working.  Looking at the code, that function is in there, but I'm wondering if there's some kind of initialization race condition that didn't occur in old JavaScript but now we're hitting it?

I have updated the "rough sample" HTML/JS code, and it works.  Just not pretty.  No GUI to configure the broker connection, etc.  You have to modify the code to make it run in your env.  Thoughts?

I will configure and add a SolCache instance to my Singapore test broker, so you can try it out easily without having to stand one up yourself.

aaron-613 committed:
trying some fixes

aaron-613 committed:
edits

Mrc0113 commented:
Closing as stale.
